src/vs/workbench/contrib/files/common/explorerModel.ts

```diff
@@ -149,7 +149,7 @@ export class ExplorerItem {
 	}
 
 	get isReadonly(): boolean {
-		return this.filesConfigService.isReadonly(this.resource, { resource: this.resource, name: this.name, readonly: this._readonly, locked: this._locked });
+		return !!this.filesConfigService.isReadonly(this.resource, { resource: this.resource, name: this.name, readonly: this._readonly, locked: this._locked });
```

Comment:
I wonder if we could get the message also into the file explorer as a hover to the readonly decoration?

Reply:
I think we could, but who would provide the read-only message? With this PR, the read-only message can come from extension file system providers and an extension wouldn't be able to set the read-only message of an explorer item.

Reply:
Started looking into this, but for that hover we need a string, not a markdown string. We can use the plain text markdown renderer, but that introduces a layering problem. I am happy to look into this further, but I would suggest that we do so in another PR to keep the scope narrow.

Reply:
Ok works for me. Should probably have a follow up issue then.

Reply:
Sure, created https://github.com/microsoft/vscode/issues/185322